File: android-beans/src/main/java/rocks/frieler/android/beans/BeanRegistryPostProcessor.java
package rocks.frieler.android.beans;

/**
 * Callback-interface that allows to post-process a {@link BeanRegistry} after all beans are collected.
 * <p>
 * Implementers of this interface that are registered in a {@link BeanRegistry} are applied to it automatically.
 *
 * @author Christopher Frieler
 */
public interface BeanRegistryPostProcessor {
    void postProcess(BeanRegistry beanRegistry);
}

File: android-beans/src/main/kotlin/rocks/frieler/android/beans/BeanRegistryPostProcessor.kt
package rocks.frieler.android.beans

/**
 * Callback-interface that allows to post-process a [BeanRegistry] after all beans are collected.
 *
 * Implementors of this interface that are registered in a [BeanRegistry] are applied to it
 * automatically.
 *
 * @author Christopher Frieler
 */
fun interface BeanRegistryPostProcessor {
    /**
     *
     */
    fun postProcess(beanRegistry: BeanRegistry)
}
